ui/apps/profile/src/components/profile-header/withProfileHeader.tsx

```diff
@@ -0,0 +1,20 @@
+import React, { ReactNode, Suspense } from 'react';
+import FollowingFeedback from '../following-feedback';
+import ProfileHeaderView, { ProfileHeaderViewProps } from './profile-header';
+import { ProfileHeaderLoading } from '@akashaorg/design-system-components/lib/components/Profile';
+
+export const withProfileHeader = <T extends ProfileHeaderViewProps>(
```

Comment:
In this case it's sole purpose is to make the code readable. It's used inside `app-routes` like this: 
```
  <Route
              path={`:profileId${menuRoute[FOLLOWING]}`}
              element={withProfileHeader(<FollowingPage showLoginModal={showLoginModal} />)(
                commonHeaderViewProps,
              )}
            />
  ```

Reply:
hmm, I don't think it makes it more readable, imagine a component that has many more props:

```
withProfileHeader(
    <SomeComponent
       with={many}
       more={props}
    />
)(
  someOtherProps
)
```
do you think is more readable than
```
  <ProfileHeader someProps={}>
    <SomeComponent
       with={many}
       more={props}
    />
  </ProfileHeader>
  ```
  ?

Reply:
```
<ProfileHeader someProps={}>
    <SomeComponent
       with={many}
       more={props}
    />
  </ProfileHeader>
  ```
  
  Profile header containing siblings as children is something I wasn't able to wrap my head around that's why I created the HOC :D

Reply:
using fragments like:

```
<>
  <ProfileHeader />
  <OtherComponent />
</>
```
or actually modifying ProfileHeader to accept children (because it suppose to be a header)